### PR TITLE
Improve internal package tests

### PR DESCRIPTION
**Pull request by kevindiu**
### Description:

This PR include the following changes:
- using NOP logger for test
- wrap goleak library to internal package, default ignore conflict with parallel test
- update goleak import path
- Update exist tests to add prefix to the environment name
- fix internal/db/storage/blob/s3/reader/reader_test.go error
- fix internal/file/file_test.go goleak error
- close watcher after test in internal/file/watch/watch_test.go

The PR seems huge but most of them are updating NOP logger for test, and update the goleak import path.

Since goleak doesn't support parallel test, this PR does not update to support parallel test.
ref: https://github.com/uber-go/goleak/issues/16

### Environment:

- Go Version: 1.16
- Docker Version: 19.03.8
- Kubernetes Version: 1.18.2
- NGT Version: 1.12.3

### Types of changes:

- [ ] Bug fix [type/bug]
- [ ] New feature [type/feature]
- [ ] Add tests [type/test]
- [ ] Security related changes [type/security]
- [ ] Add documents [type/documentation]
- [x] Refactoring [type/refactoring]
- [ ] Update dependencies [type/dependency]
- [ ] Update benchmarks and performances [type/bench]
- [ ] Update CI [type/ci]

### Changes to Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?

### Checklist:

- [x] I have read the [CONTRIBUTING](https://github.com/vdaas/vald/blob/master/CONTRIBUTING.md) document.
- [x] I have checked open [Pull Requests](https://github.com/vdaas/vald/pulls) for the similar feature or fixes?
- [ ] I have added tests and benchmarks to cover my changes.
- [x] I have ensured all new and existing tests passed.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: internal/file/watch/watch_test.go [hunk 1]
```diff
@@ -436,7 +488,8 @@ func Test_watch_Start(t *testing.T) {
 						},
 					}
 				},
-				afterFunc: func(args) {
+				afterFunc: func(t *testing.T, args args, w Watcher) {
```
[the remark this answers is not in the document]
Comment (kevindiu) [reply internal/file/watch/watch_test.go:488]:
```suggestion
				afterFunc: func(t *testing.T, args args, w Watcher) {
					t.Helper()
```

File: internal/file/watch/watch_test.go [hunk 2]
```diff
@@ -389,7 +440,8 @@ func Test_watch_Start(t *testing.T) {
...
```
[the remark this answers is not in the document]
Comment (kevindiu) [reply internal/file/watch/watch_test.go:440]:
```suggestion
				afterFunc: func(t *testing.T, args args, w Watcher) {
					t.Helper()
```

File: internal/file/watch/watch_test.go [hunk 3]
```diff
@@ -336,7 +385,9 @@ func Test_watch_Start(t *testing.T) {
...
```
[the remark this answers is not in the document]
Comment (kevindiu) [reply internal/file/watch/watch_test.go:385]:
```suggestion
				afterFunc: func(t *testing.T, args args, w Watcher) {
					t.Helper()
```

File: internal/file/watch/watch_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (kevindiu) [reply internal/file/watch/watch_test.go:440]:
```suggestion
				afterFunc: func(t *testing.T, args args, w Watcher) {
					t.Helper()
```

File: internal/file/watch/watch_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (kevindiu) [reply internal/file/watch/watch_test.go:488]:
```suggestion
				afterFunc: func(t *testing.T, args args, w Watcher) {
					t.Helper()
```

File: internal/file/watch/watch_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (kevindiu) [reply internal/file/watch/watch_test.go:385]:
```suggestion
				afterFunc: func(t *testing.T, args args, w Watcher) {
					t.Helper()
```

File: internal/file/watch/watch_test.go [hunk 4]
```diff
@@ -931,6 +976,14 @@ func Test_watch_Remove(t *testing.T) {
 			}
 		}
 	}
+	defaultAfterFunc := func(t *testing.T, args args, w Watcher) {
```
[the remark this answers is not in the document]
Comment (kevindiu) [reply internal/file/watch/watch_test.go:1010]:
```suggestion
	defaultAfterFunc := func(t *testing.T, args args, w Watcher) {
		t.Helper()
```

File: internal/file/watch/watch_test.go [hunk 5]
```diff
@@ -774,6 +810,14 @@ func Test_watch_Add(t *testing.T) {
 			t.Fatal(err)
 		}
 	}
+	defaultAfterFunc := func(t *testing.T, args args, w Watcher) {
```
[the remark this answers is not in the document]
Comment (kevindiu) [reply internal/file/watch/watch_test.go:833]:
```suggestion
	defaultAfterFunc := func(t *testing.T, args args, w Watcher) {
		t.Helper()
```

File: internal/db/storage/blob/s3/reader/reader_test.go [hunk 6]
```diff
@@ -422,9 +423,13 @@ func Test_reader_Open(t *testing.T) {
 								return
 							default:
 								if roopCnt == 0 {
-									_, _ = r.Read(bytes[0])
+									if _, err := r.Read(bytes[0]); err == io.EOF {
```
[the remark this answers is not in the document]
Comment (kevindiu) [reply internal/db/storage/blob/s3/reader/reader_test.go:426]:
```suggestion
									if _, err := r.Read(bytes[0]); errors.Is(err, io.EOF) {
```

File: internal/db/storage/blob/s3/reader/reader_test.go [hunk 7]
```diff
@@ -422,9 +423,13 @@ func Test_reader_Open(t *testing.T) {
...
+										return
+									}
 								} else {
-									_, _ = r.Read(bytes[1])
+									if _, err := r.Read(bytes[1]); err == io.EOF {
```
[the remark this answers is not in the document]
Comment (kevindiu) [reply internal/db/storage/blob/s3/reader/reader_test.go:430]:
```suggestion
									if _, err := r.Read(bytes[1]); error.Is(err, io.EOF) {
```

File: internal/file/file_test.go [hunk 8]
```diff
@@ -61,6 +61,14 @@ func TestOpen(t *testing.T) {
 		}
 		return nil
 	}
+	defaultAfterFunc := func(t *testing.T, args args, f *os.File) {
+		t.Helper()
+		if f != nil {
```
[the remark this answers is not in the document]
Comment (kevindiu) [reply internal/file/file_test.go:66]:
```suggestion

		if f != nil {
```

File: internal/file/watch/watch_test.go [hunk 9]
```diff
@@ -149,6 +165,23 @@ func Test_watch_init(t *testing.T) {
...
+			}
+			if watch, ok := w.(*watch); ok {
+				if watch.w != nil {
+					err := watch.w.Close()
+					if err != nil {
+						t.Error(err)
+					}
+				}
+			}
+		}
```
Comment (kevindiu) [inline internal/file/watch/watch_test.go:183]:
```suggestion
		if w == nil {
			return
		}
		
		err := w.Stop(context.Background())
		if err != nil {
			t.Error(err)
		}
		if watch, ok := w.(*watch); ok {
			if watch.w != nil {
				err := watch.w.Close()
				if err != nil {
					t.Error(err)
				}
			}
		}
```

File: internal/file/watch/watch_test.go [hunk 10]
```diff
@@ -482,7 +537,8 @@ func Test_watch_Start(t *testing.T) {
 						},
 					}
 				},
-				afterFunc: func(args) {
+				afterFunc: func(t *testing.T, args args, w Watcher) {
```
[the remark this answers is not in the document]
Comment (kevindiu) [reply internal/file/watch/watch_test.go:540]:
```suggestion
				afterFunc: func(t *testing.T, args args, w Watcher) {
					t.Helper()
```

File: internal/file/watch/watch_test.go [hunk 11]
```diff
@@ -528,7 +584,8 @@ func Test_watch_Start(t *testing.T) {
...
```
[the remark this answers is not in the document]
Comment (kevindiu) [reply internal/file/watch/watch_test.go:587]:
```suggestion
				afterFunc: func(t *testing.T, args args, w Watcher) {
					t.Helper()
```

File: internal/file/watch/watch_test.go [hunk 12]
```diff
@@ -574,7 +631,8 @@ func Test_watch_Start(t *testing.T) {
...
```
[the remark this answers is not in the document]
Comment (kevindiu) [reply internal/file/watch/watch_test.go:634]:
```suggestion
				afterFunc: func(t *testing.T, args args, w Watcher) {
					t.Helper()
```

**Comment by kpango** [comment]
it would be better to update header.tmpl to change import path for goleak

**Comment by kevindiu** [comment]
Should we use parallel test but implement the following function in each package
```go
func TestMain(m *testing.M) {
	goleak.VerifyTestMain(m)
}
```
or should we disable parallel test?

**Review by hlts2 — DISMISSED** [review]

**Review by hlts2 — DISMISSED** [review]

**Review by hlts2 — APPROVED** [review]

**Review by kpango — APPROVED** [review]
LGTM

**Review by vankichi — APPROVED** [review]

**Pull request merged**